Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -6,13 +6,81 @@ import Common
 import ComponentLibrary
 import UIKit
 import Shared
+import Redux
+
+struct FakespotState: ScreenState, Equatable {
+    var isSettingsExpanded: Bool
+    var isReviewQualityExpanded: Bool
+    let logger: Logger = DefaultLogger.shared
+
+    enum Action: Redux.Action {
+        case settingsStateDidChange
+        case reviewQualityDidChange
+        case urlDidChange
+    }
+
+    init() {
+        self.isSettingsExpanded = false
+        self.isReviewQualityExpanded = false
+    }
+
+    init(isSettingsExpanded: Bool, isReviewQualityExpanded: Bool) {
+        self.isSettingsExpanded = isSettingsExpanded
+        self.isReviewQualityExpanded = isReviewQualityExpanded
+    }
+
+    init(_ appState: AppState) {
+        guard let state = store.state.screenState(FakespotState.self, for: .fakespot) else {
+            self.init(isSettingsExpanded: false, isReviewQualityExpanded: false)
+            logger.log("Error retrieving screen state",
+                       level: .debug,
+                       category: .redux)
+            return
+        }
+
+        self.init(
+            isSettingsExpanded: state.isSettingsExpanded,
+            isReviewQualityExpanded: state.isReviewQualityExpanded
+        )
+    }
+
+    static let reducer: Reducer<Self> = { state, action in
+        switch action {
+        case Action.settingsStateDidChange:
```

Comment:
I would have loved to have type inference here, so I would make Reducer generic over both `State` and `Action`. One downside is no autocomplete, and for future actions we will not get a compiler error because we always need the default case.

Reply:
🤔 this will tight `State` and `Action` together which is the case most of the time, let me think of other scenarios where this could be an issue. Also I will love @OrlaM feedback on a lot of this topic so I will tag her in the PR

Reply:
Can you elaborate on what change you are suggesting here?

Reply:
Yes, so `action` here is just a general any Action protocol no information about its type. When you switch on it you lose the autocomplete because it has lost its type, so that's why you have to add `Type.`. And if you add another action the compiler will be like fine, the code is building, but there's not indication that you have not handled that case. So I think that Action should be generic to not lose its type.